File: 05_basic_recursion/02_reverse_array/solution.java
// https://www.geeksforgeeks.org/problems/reverse-an-array/0
class Solution {
    public void reverseArray(int arr[]) {
        reverseArrayRecursive(arr, 0, arr.length-1);
    }
    
    public void reverseArrayRecursive(int arr[], int left, int right) {
        if (left >= right) {
            return;
        }
        
        // swap value at index left and index right
        int temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
        
        reverseArrayRecursive(arr, left+1, right-1);
    }
}

File: 05_basic_recursion/02_reverse_array/solution.py
# https://www.geeksforgeeks.org/problems/reverse-an-array/0
class Solution:
    def reverseArray(self, arr):
        self.reverseArrayRecursive(arr, 0, len(arr)-1)
    
    def reverseArrayRecursive(self, arr, left: int, right: int):
        if left >= right:
            return
        
        arr[left], arr[right] = arr[right], arr[left]
        
        return self.reverseArrayRecursive(arr, left+1, right-1)